indolaw-parser/parser_utils.py

```diff
@@ -318,56 +318,156 @@ def clean_law(law: List[str]) -> List[str]:
     law = [line.strip() for line in law]
     law = [' '.join(line.split()) for line in law]
 
-    save_law_to_file(law, 'a.txt')
-
     '''
-    Remove semantically meaningless text e.g '. . .' or '1 / 23'
-
-    Mostly, they come in whole lines, but sometimes they get squashed
-    onto the end of real lines
-
-    e.g a real line ending in '2 / 43' in UU 18 2017
+    Preps for cleaning stages for potential save in a dict of stages
+    TODO(Mel):  1. Maybe change law_stages headers into enums?
+                2. Extend saving mechanism to catch and save during a crash
+                3. Add documentation on how saving mechanism works 
     '''
-    law = list(filterfalse(ignore_line, law))
-    law = clean_squashed_page_numbers(law)
+    law_stages = [['clean_squashed_page_numbers',[]], ['clean_maybe_list_items',[]],
+                    ['clean_maybe_squashed_headings',[]], ['clean_split_plaintext',[]],
+                    ['clean_split_pasal_number',[]], ['insert_perubahan_quotes',[]]]
+    cleaning_progress = {'started': False, 'finished': False}
+    
+    while cleaning_progress['finished'] == False:
+        
+        user_input_stage = '0'
+
+        if cleaning_progress['started'] == True:
+
+            print("---------------")
```

Comment:
Yeah, noticed this after I requested the PR 😄 (because I was using the parser code from last week before pulling from origin today). Will change it today!